src/components/Chat/InputMessageV2.tsx

```diff
@@ -119,19 +140,202 @@ const InputMessageV2 = ({
+        multiple: true,
+        maxFiles: 20,
+        sortOrder: 'asc',
+        smartAlbums: ['RecentlyAdded', 'UserLibrary']
+      }).then(async (medias) => {
+        const newMedias: any = [];
+        for (const media of medias) {
+          if (media?.mime?.includes('video')) {
+            const thumbnailPath = await getVideoThumbnail(media.path);
+            newMedias.push({
+              type: 'video',
+              path: thumbnailPath,
+              pathVideo: media.path,
+              nameVideo: media.filename,
+              typeVideo: media.mime
+            });
+          } else {
+            const imageCropped = await ImagePicker.openCropper({
+              mediaType: 'photo',
+              path: media.path,
+              width: media.width,
+              height: media.height,
+              cropperChooseText: 'Next',
+              freeStyleCropEnabled: true
+            });
+            newMedias.push({type: 'image', path: imageCropped.path});
+          }
+        }
+
+        handleUploadMedia(newMedias);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your photos, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenCamera = async () => {
+    const {success} = await requestCameraPermission();
+    if (success) {
+      ImagePicker.openCamera({
+        mediaType: 'photo'
+      }).then(async (imageRes) => {
+        const imageCropped = await ImagePicker.openCropper({
+          mediaType: 'photo',
+          path: imageRes.path,
+          width: imageRes.width,
+          height: imageRes.height,
+          cropperChooseText: 'Next',
+          freeStyleCropEnabled: true
+        });
+        handleUploadCamera(imageCropped.path);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your camera, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenFile = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      DocumentPicker.pickSingle({
+        presentationStyle: 'fullScreen'
+      }).then((pickerResult) => {
+        handleFile(pickerResult);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your document library, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
 
   const onChangeInput = (v) => {
     setText(v);
   };
-  const onSelectEmoji = () => {
-    refEmoji.current.close();
+
+  const onOpenGIF = () => {
+    refAttachment.current.close();
+    setTimeout(() => {
+      refGif.current.open();
+    }, 500);
+  };
+
+  const onCloseGIF = () => {
+    refGif.current.close();
+
+
+  const onSelectAttachment = () => {
+    refAttachment.current.open();
   };
 
   const handleSendMessage = () => {
-    onSendButtonClicked(text);
+    onSendButtonClicked(text, []);
+    onSendButtonClicked(text, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This hunk introduces a lot of new functionality related to handling media attachments in chat messages. There are a few issues that need to be addressed:

1. Error handling: The new functions `handleUploadMedia`, `handleUploadCamera`, `handleSelectGIF`, `handleFile`, `getVideoThumbnail`, `onOpenMedia`, `onOpenCamera`, and `onOpenFile` all involve asynchronous operations but do not have proper error handling. It's important to add try-catch blocks around these operations to handle any potential errors and provide a good user experience.

2. Code duplication: The `handleUploadMedia` and `handleUploadCamera` functions both call `onSendButtonClicked` with similar arguments and then set their respective loading states to false. This duplicated logic could be extracted into a separate function.

3. Redundant code: On lines 151 and 337, `onSendButtonClicked` is called twice consecutively with the same arguments. One of these calls should be removed.

4. Use of console.warn: On line 209, there's a `console.warn` statement. It's generally not a good practice to leave console statements in production code as they can expose sensitive information.



<!-- This is an auto-generated comment by OSS CodeRabbit -->